### ISSUE TITLE
Import Categories fail in 1 action

### ISSUE THREAD

**Issue by thanhtantran** [Agence-DnD/PIMGento-2#197]
I try to test import categories and get this error

 bin/magento pimgento:import --code=category --file=categories.csv
[09:31:29] Start import
[09:31:29] Import id: 5b7a8a71a83d1
[09:31:29] Create temporary table
[09:31:29] completed
[09:31:29] Fill temporary table
[09:31:29] SQLSTATE[28000]: Invalid authorization specification: 1045 Access denied for user 'zzzmwxsvnj'@'%' (using password: YES), query was: LOAD DATA INFILE '/tmp/categories.csv' REPLACE
              INTO TABLE tmp_pimgento_entities_category
              CHARACTER SET UTF8
              FIELDS TERMINATED BY ';'
              OPTIONALLY ENCLOSED BY '"'
              LINES TERMINATED BY '\n'
              IGNORE 1 LINES;

**Comment by lbassin** [comment]
Hi @thanhtantran 

It seems that your user isn't allowed to use the function "LOAD DATA INFILE". Some servers disabled it by default

In order to fix this, you may try to change LOAD DATA configuration
- Go to magento dashboard. Stores > Configuration then Catalog > Pimgento
- In the General section you'll find an option "Load Data Local" set to "No". Change this setting to "Yes"
- Save and re-try to import data

If the error remains, try to change pimgento data insertion method this way
- Go to magento dashboard. Stores > Configuration then Catalog > Pimgento
- In the General section you'll find an option "Data insertion method" set to "Data in file". Change this setting to "By rows"
- Save and re-try to import data

Note that "By rows" method is slower than "Data in file"

**Comment by thanhtantran** [comment]
Thank you. I have successful import "by row", however that data from Akeneo is not compatible with Magento 2. This cause the categories imported without any information

My sample categories export from Akaneo bellow

[code;label-en_US;parent
master;"Master catalog";
002;"Văn Học Nước Ngoài";master
021;"Hiện đại";002
022;"Hồi ký - Tự truyện";002
023;"Kiếm hiệp";002](url)

**Comment by lbassin** [comment]
What do you mean by "imported without any information" ? 
I tried to import the data your provided and it works

![image](https://user-images.githubusercontent.com/11029822/44709294-90c7ac80-aaa9-11e8-9e41-789cc0cead93.png)

![image](https://user-images.githubusercontent.com/11029822/44709396-ccfb0d00-aaa9-11e8-82aa-2b5c37145181.png) ![image](https://user-images.githubusercontent.com/11029822/44709306-99b87e00-aaa9-11e8-9abe-bd33ff7f9c29.png)

**Comment by thanhtantran** [comment]
I mean the Categories after import appear as this

[image: Screenshot_1.jpg]
---
Tran Thanh Tan
Mobile: 090.444.9899
Backup: 092.344.9899

**Comment by thanhtantran** [comment]
I don't know how to display image.

@lbassin : Do I need to configure anything in Configurations?

**Comment by lbassin** [comment]
I don't think there is a specific configuration to change
Have you tried removing the empty categories and importing them again ? 

I'll need more information in order to fix this issue
What do you have in the pimgento entities table ? You may find this information with the sql query: 
```sql
select * from pimgento_entities where import = 'category'
```

**Comment by thanhtantran** [comment]
Yes. checked and already have pimgento_entities  table. I have tried empty it re import but no sucess.

Maybe the problem is from Magento version. I tried the same categories export with 2.10, it works.

It not work with 2.25

**Comment by lbassin** [comment]
I don't think version has something to do in this issue, I successfully imported your data in a 2.2.5 magento installation

Have a closer look to the pimgento_entities table.
It is in charge to map akeneo entity to magento entity in order to avoid duplicate data
Maybe there was a problem in a previous import that corrupted data and now pimgento failed to match entities and import empty categories

**Comment by thanhtantran** [comment]
I have installed a new Magento them import by Pimgento, that works very good. 

But the second import when i want to add more categories or update the name of categories, it display nothing. Only (0) for the new added categories

![screenshot_1](https://user-images.githubusercontent.com/5319910/45413611-cf618780-b6a3-11e8-8dea-023016f3c4bb.jpg)

**Comment by lbassin** [comment]
Mhm this is a strange behavior. I'll try to repeat the same steps and see what happens

What categories did you import with the second file ? A mix of new and old ? 
Does the update work on existing categories ?
